### PR TITLE
[Netatmo] Added NPE safe-guard

### LINKED ISSUE #4263: [Netatmo] NPE while updating channels for devices
Results in a complete blocking of the whole binding. No more new data is retrieved from API.

See https://community.openhab.org/t/netatmo-relay-and-thermostat/36463/28

```
2018-11-20 15:25:28.266 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.NullPointerException: null
        at org.openhab.binding.netatmo.internal.station.NAMainHandler.getNAThingProperty(NAMainHandler.java:63) ~[?:?]
        at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.lambda$3(AbstractNetatmoThingHandler.java:109) ~[
?:?]
        at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) ~[?:?]
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) ~[?:?]
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471) ~[?:?]
        at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:?]
        at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418) ~[?:?]
        at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.updateChannels(AbstractNetatmoThingHandler.java:1
06) ~[?:?]
        at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.updateChannels(NetatmoDeviceHandler.java:157) ~[?:?]
        at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.lambda$0(NetatmoDeviceHandler.java:84) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:
?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
```

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added NPE safe-guard
- Fixed SAT findings

Closes #4263 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java [hunk 1]
```diff
@@ -86,31 +86,31 @@ public static State toOnOffType(@Nullable Integer value) {
         return value != null ? (value == 1 ? OnOffType.ON : OnOffType.OFF) : UnDefType.NULL;
     }
 
-    public static State toQuantityType(@Nullable Float value, @Nullable Unit<?> unit) {
-        return (value == null || unit == null) ? UnDefType.NULL : toQuantityType(new BigDecimal(value), unit);
+    public static State toQuantityType(@Nullable Float value, Unit<?> unit) {
+        return (value == null) ? UnDefType.NULL : toQuantityType(new BigDecimal(value), unit);
```
Comment (Hilbrand) [inline addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java]: Maybe remove the parenthesis around the check too as it's only 1 check.

**Comment by paulianttila** [comment]
@cweitkamp, I tested the fix (build: https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.netatmo/2.4.0-SNAPSHOT/), but it seems that it doesn't fully fix the problem (at least not on my environment).
 
`2018-11-22 19:23:46.938 [ERROR] [ome.core.thing.link.ThingLinkManager] - Exception occurred while informing handler: null
java.lang.NullPointerException: null
	at org.openhab.binding.netatmo.internal.station.NAModule1Handler.getNAThingProperty(NAModule1Handler.java:49) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.lambda$3(AbstractNetatmoThingHandler.java:109) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.updateChannels(AbstractNetatmoThingHandler.java:106) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.NetatmoModuleHandler.updateChannels(NetatmoModuleHandler.java:97) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.lambda$7(NetatmoDeviceHandler.java:202) ~[?:?]
	at java.util.Optional.ifPresent(Optional.java:159) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.lambda$5(NetatmoDeviceHandler.java:200) ~[?:?]
	at java.util.concurrent.ConcurrentHashMap.forEach(ConcurrentHashMap.java:1597) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.updateChildModules(NetatmoDeviceHandler.java:198) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.NetatmoDeviceHandler.updateChannels(NetatmoDeviceHandler.java:158) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.handleCommand(AbstractNetatmoThingHandler.java:133) ~[?:?]
	at org.eclipse.smarthome.core.thing.binding.BaseThingHandler.channelLinked(BaseThingHandler.java:278) ~[?:?]
	at org.openhab.binding.netatmo.internal.handler.AbstractNetatmoThingHandler.channelLinked(AbstractNetatmoThingHandler.java:119) ~[?:?]
	at org.eclipse.smarthome.core.thing.link.ThingLinkManager.lambda$0(ThingLinkManager.java:300) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
`

Full log: [netatmo.txt](https://github.com/openhab/openhab2-addons/files/2608993/netatmo.txt)

**Comment by cweitkamp** [comment]
@paulianttila Thanks for your feedback. I am afraid the linked jar file is either not the latest  version or does not contain my changes. The NPE in your log occurred in line 49 of the `NAModule1Handler.java`. This line contains a `case` and thus - please correct me if I am wrong - cannot throw a NPE.

https://github.com/openhab/openhab2-addons/blob/00904041758634d764f62246b00cfabf0ba565b7/addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/station/NAModule1Handler.java#L47-L51.

File: addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java (hunk 1 above)
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java]: I did it.

**Comment by wborn** [comment]
@paulianttila you have probably downloaded/tested a Netatmo JAR generated by a build for another PR. See also the Note in [Distributing bindings through the IoT Marketplace](https://community.openhab.org/t/distributing-bindings-through-the-iot-marketplace/24491?u=wborn).

> Note: The shared Maven repo for PRs obviously has the risk that if somebody else includes changes to a certain binding in his PR that the binary jar is changed accordingly - so it can be tricky to tell, what its origin is. I therefore only recommend using these urls for new bindings that are not yet part of the repo - refactorings of existing bindings are probably better offered elsewhere.

**Pull request merged**